pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -1270,6 +1270,11 @@ public void addTable(TableConfig tableConfig)
         if (getAllTables().contains(tableNameWithType)) {
           throw new TableAlreadyExistsException("Table " + tableNameWithType + " already exists");
         }
+        List<SegmentZKMetadata> segmentsZKMetadata = getSegmentsZKMetadata(tableNameWithType);
+        if (segmentsZKMetadata != null && segmentsZKMetadata.size() != 0) {
+          throw new TableAlreadyExistsException("Table " + tableNameWithType + " still have active segments registered"
```

Comment:
Table already exists may not be the right one, depending on whether the table is under deletion now? Is it guaranteed that the delete call removes all metadata right away?

Reply:
nope. it doesn't, however I can't find a better exception type. should I create a new one?

Reply:
you can change the exception message accordingly.  Provide as accurate a message as possible